src/main/java/org/apache/commons/validator/routines/checkdigit/VATidBGCheckDigit.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.validator.routines.checkdigit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.validator.GenericTypeValidator;
+import org.apache.commons.validator.GenericValidator;
+import org.apache.commons.validator.routines.DateValidator;
+
+/**
+ * Bulgarian VAT identification number (VATIN) Check Digit calculation/validation.
+ * <p>
+ * Dank dobawena stoinost (DDS)
+ * The Bulgarian VAT (Данък върху добавената стойност) number is either 9 (for legal entities)
+ * or 10 digits long (ЕГН for physical persons, foreigners and others).
+ * Each type of number has its own check digit algorithm.
+ * </p>
+ * <p>
+ * See <a href="https://en.wikipedia.org/wiki/VAT_identification_number">Wikipedia VATIN</a>
+ * and <a href="https://en.wikipedia.org/wiki/Unique_citizenship_number">ЕГН (civil number)</a>
+ * for more details.
+ * </p>
+ *
+ * @since 1.10.0
+ */
+public final class VATidBGCheckDigit extends ModulusCheckDigit {
+
+    private static final long serialVersionUID = -8667365895223831325L;
+    private static final Log LOG = LogFactory.getLog(VATidBGCheckDigit.class);
+
+    /** Singleton Check Digit instance */
+    private static final VATidBGCheckDigit INSTANCE = new VATidBGCheckDigit();
+
+    /**
+     * Gets the singleton instance of this validator.
+     * @return A singleton instance of the class.
+     */
+    public static CheckDigit getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * there ate three length
```

Comment:
Sentences start with a capital letter. If you expect the comment below to be renders line by line, use a `ol` or `ul` HTML tag.